Extensions/read_more_now.js

```diff
@@ -1,211 +1,121 @@
-					XKit.extensions.read_more_now.show_failed();
-				},
-				onload: function(response) {
-					var data = JSON.parse(response.responseText);
-					var m_object = data.response;
-					try {
-						var m_contents = m_object.posts[0].body;
-
-						if (m_object.posts[0].type === "photo") {
-							m_contents = m_object.posts[0]["photo-caption"] ||
-						             m_object.posts[0].caption;
-						}
-
-						if (m_object.posts[0].type === "answer") {
-							m_contents = m_object.posts[0].answer;
-						}
-
-						if (m_object.posts[0].type === "link") {
-							m_contents = m_object.posts[0].description;
-						}
-
-						var post_cont = $(m_cont).parent().parent();
-						if (post_cont.find(".post_title").length > 0) {
-							var post_title = post_cont.find(".post_title")[0].outerHTML;
-							post_cont.html(XKit.extensions.read_more_now.strip_scripts(post_title + m_contents));
-						} else {
-							post_cont.html(XKit.extensions.read_more_now.strip_scripts(m_contents));
-						}
-
-						if (XKit.interface.where().search) {
-							post_cont.find("img").load(function() {
-								XKit.interface.trigger_reflow();
-							});
-							XKit.interface.trigger_reflow();
-						}
-					} catch (e) {
-						$(m_cont).removeClass("disabled");
-						$(m_cont).html(XKit.extensions.read_more_now.button_caption);
-						XKit.extensions.read_more_now.show_failed();
-					}
-				}});
+		$(document).on("click", ".xreadmore:not(.disabled)", function() {
+			XKit.extensions.read_more_now.add_content($(this));
 		});
 	},
 
-	strip_scripts: function(post_html) {
-
-		/*
-			From:
-			http://stackoverflow.com/questions/6659351/removing-all-script-tags-from-html-with-js-regular-expression
-		*/
-
-		var div = document.createElement('div');
-		div.innerHTML = post_html;
-		var scripts = div.getElementsByTagName('script');
-		var i = scripts.length;
-		while (i--) {
-			scripts[i].parentNode.removeChild(scripts[i]);
-		}
-		return div.innerHTML;
-	},
-
-	show_failed: function() {
-
-		XKit.window.show("Unable to fetch read more", "Perhaps the user deleted the post?", "error", "<div class=\"xkit-button default\" id=\"xkit-close-message\">OK</div>");
-
+	find_links: function() {
+		$("a.tmblr-truncated-link:not(.xreadmore-done)")
+			.addClass("xreadmore-done")
+			.each(function() {
+				XKit.extensions.read_more_now.add_button($(this));
+			});
 	},
 
-	do: function() {
-		var need_reflow = false;
+	add_button: function($link) {
+		const href = $link.attr("href");
+		let url;
+		let postID;
 
-		$(".posts .post").not(".xread-more-now-done").each(function() {
-			var post = $(this);
-			if (post.hasClass("xread-more-now-done")) {
-				return;
-			}
-			post.addClass("xread-more-now-done");
-
-			if (post.hasClass("read_more_container")) {
-				XKit.extensions.read_more_now.handle_read_more_post(post);
-				need_reflow = true;
+		if (!href.includes("t.umblr.com")) {
+			[url,, postID] = href.split("://")[1].split("/");
```

Comment:
I'm guessing this isn't a typo but adding a `_` to the middle would make your intent clearer

Reply:
i thought this was less ugly than either using `__` or taking another line to `let _;` since just adding `_` here overwrites the entirety of lodash lol

happy to change it, the question is to which option

Reply:
Riiiight, of course this isn't straightforward. Maybe `[url, /* discard */, postId]`? I just want something since my keyboard occasionally doubles the`,` key and this looks exactly like that

Reply:
ooooh yeah, that's good